Automattic/vip-go-mu-plugins
# Add filter for vary cache

## Description

The change adds a filter to allow modifying `Vary:` header.



## Changelog Description

### Add `vip_vary_header` filter to modify `Vary` header value.

The change adds a filter to allow modifying `Vary:` header

Usage example https://cleanshot.thrijith.com/kBB5yq55


## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
2. Add filter in your client-mu-plugins example.

```
/**
 *  Modify vary cache header.
 * 
 * @param string $vary_header The vary header that VIP's vary cache sets.
 */
function update_vary_header( $vary_header ) {
	return "$vary_header, Origin";
}
add_filter( 'vip_vary_header', __NAMESPACE__ . '\\update_vary_header', 10 );
```
3. Do a cURL request to the site which you added filter or check in browser, confirm you can see the change.
4. Remove the filter, without the filter there should be no change to the existing header.
